backend/controller/observability/fsm.go

```diff
@@ -2,53 +2,91 @@ package observability
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/noop"
 
 	"github.com/TBD54566975/ftl/backend/schema"
 	"github.com/TBD54566975/ftl/internal/observability"
 )
 
 const (
-	fsmMeterName    = "ftl.fsm"
-	fsmRefAttribute = "ftl.fsm.ref"
+	fsmMeterName             = "ftl.fsm"
+	fsmRefAttribute          = "ftl.fsm.ref"
+	fsmDestStateRefAttribute = "ftl.fsm.dest.state.ref"
 )
 
-var fsmMeter = otel.Meter("ftl.fsm")
+type FSMMetrics struct {
+	meter              metric.Meter
+	instancesActive    metric.Int64UpDownCounter
+	transitionsActive  metric.Int64UpDownCounter
+	transitionAttempts metric.Int64Counter
+}
 
-var fsmCounters = struct {
-	instancesActive metric.Int64UpDownCounter
-}{}
+func initFSMMetrics() (*FSMMetrics, error) {
+	result := &FSMMetrics{}
 
-func InitFSMMetrics() error {
+	var errs error
 	var err error
```

Comment:
Would you mind removing this `var err`? Generally, you only initialize vars in Go when you have to (e.g. to get an address of a particular type), so for this, we can just use `:=` for the first `result.blah, err := `

Reply:
Ah nvm, I see where you need it now